File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        for(int i = 0; i < win_nums.length; i++){
            for(int j = 0; j < lottos.length; j++){
                if(win_nums[i] == lottos[j]) win_nums[i] = -1;
            }
        }
        
        // hit는 현재 맞은 수를 계산한 것
        int hit = 0;
        for(int i = 0; i < win_nums.length; i++){
            if(win_nums[i] == -1) hit++;
        }
        
        int pot = 0;
        for(int i = 0; i < lottos.length; i++) if(lottos[i] == 0) pot++;
        answer[0] = 7 - hit - pot;
        answer[1] = 7 - hit;
        
        if(answer[0] == 0) answer[0]++;
        if(answer[0] == 7) answer[0]--;
        if(answer[1] == 7) answer[1]--;
        
        return answer;
    }
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    let answer = [];
    let potential = 0;
    let correct = 0;
    
    for(lotto of lottos){
        if(lotto == 0) potential++;
        for(win of win_nums){
            if(lotto == win) correct++;
        }
    }
    
    answer[0] = correct + potential > 1 ? 7 - (correct + potential) : 6;
    answer[1] = correct > 1 ? 7 - correct : 6;
    return answer;
}